catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/GmdTransformer.java

```diff
@@ -210,187 +306,596 @@ private void copyArgumentsToContext(MarshallingContext context,
+
+        addFieldsToMetacardIfPresent(pathValueTracker,
+                metacard,
+                GmdConstants.CONTACT_EMAIL_PATH,
+                Contact.PUBLISHER_EMAIL);
+
+        List<String> deliveryPoints =
+                pathValueTracker.getAllValues(toPath(GmdConstants.CONTACT_ADDRESS_DELIVERY_POINT_PATH));
+        if (CollectionUtils.isNotEmpty(deliveryPoints)) {
+            List<String> cities =
+                    pathValueTracker.getAllValues(toPath(GmdConstants.CONTACT_ADDRESS_CITY_PATH));
+            List<String> administrativeAreas =
+                    pathValueTracker.getAllValues(toPath(GmdConstants.CONTACT_ADDRESS_ADMINISTRATIVE_AREA_PATH));
+            List<String> postalCodes =
+                    pathValueTracker.getAllValues(toPath(GmdConstants.CONTACT_ADDRESS_POSTAL_CODE_PATH));
+            List<String> countries =
+                    pathValueTracker.getAllValues(toPath(GmdConstants.CONTACT_ADDRESS_COUNTRY_PATH));
+            List<String> addresses = new ArrayList<>();
+            for (int i = 0; i < deliveryPoints.size(); i++) {
+                StringBuilder address = new StringBuilder(deliveryPoints.get(i));
+                address.append(getStringFromListAtIndexIfNotEmpty(cities, i));
+                address.append(getStringFromListAtIndexIfNotEmpty(administrativeAreas, i));
+                address.append(getStringFromListAtIndexIfNotEmpty(postalCodes, i));
+                address.append(getStringFromListAtIndexIfNotEmpty(countries, i));
+                addresses.add(address.toString()
+                        .trim());
+            }
+            metacard.setAttribute(Contact.PUBLISHER_ADDRESS, (Serializable) addresses);
+        }
     }
 
-    private void addMetacardType(final XstreamPathValueTracker pathValueTracker,
+    private void addMetacardAltitude(final XstreamPathValueTracker pathValueTracker,
             MetacardImpl metacard) {
+        List<String> altitudeList =
+                pathValueTracker.getAllValues(toPath(GmdConstants.ALTITUDE_PATH));
+        if (CollectionUtils.isNotEmpty(altitudeList)) {
+            addMetacardAltitudeList(altitudeList, metacard);
+        }
+    }
 
-        String type = pathValueTracker.getFirstValue(toPath(GmdMetacardType.CODE_LIST_VALUE_PATH));
-        if (StringUtils.isNotEmpty(type)) {
-            metacard.setContentTypeName(type);
+    private void addMetacardAltitudeList(List<String> serviceExAltitude, MetacardImpl metacard) {
+        List<Double> altitudes = covertStringListToDoubleList(serviceExAltitude);
+        if (CollectionUtils.isNotEmpty(altitudes)) {
+            metacard.setAttribute(Location.ALTITUDE, (Serializable) altitudes);
         }
+    }
 
+    private void addMetacardTemporalStartStop(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard) {
+        List<String> temporalStartList =
+                pathValueTracker.getAllValues(toPath(GmdConstants.TEMPORAL_START_PATH));
```

Comment:
There can be multiple dateTimes that can be mapped to this field from one resource (ex. image collection time, report date, etc)